Frameworks/UIKit/UIButton.mm

```diff
@@ -371,21 +371,32 @@ - (CGRect)contentRectForBounds:(CGRect)bounds {
  @Status Interoperable
 */
 - (CGRect)imageRectForContentRect:(CGRect)contentRect {
-    CGSize titleSize = [self.currentTitle sizeWithFont:self.font];
-    // TODO  #1365 :: Currently cannot assume getting size from nil will return CGSizeZero
-    CGSize imageSize = CGSizeZero;
-    CGRect insetsRect = UIEdgeInsetsInsetRect(contentRect, self.imageEdgeInsets);
-
-    if (!self.currentImage) {
+    /////////////////////////////////////////////////////////////////////
+    // Note: #1365 All size calculations here must check for nil, 
+    // as we cannot assume getting a size from nil will return CGSizeZero
+    /////////////////////////////////////////////////////////////////////
+    UIImage* currentImage = self.currentImage;
+    if (!currentImage) {
+        // No image, so we return zero for the entire content rect
         return CGRectZero;
     }
 
-    imageSize = self.currentImage.size;
+    // Grab the size of the current title
+    // TODO: Should we be asking our label for this?
+    NSString* currentTitle = self.currentTitle;
+    CGSize titleSize = CGSizeZero;
+    if (currentTitle) {
```

Comment:
i'm saying, 'i don't know'; do we have tests to validate this behavior?  I just went through and kept the existing functionality.  The other functions check the length, and this one doesn't, because that's how it worked previously.

---
In reply to: [103035705](https://github.com/Microsoft/WinObjC/pull/2094#discussion_r103035705) [](ancestors = 103035705,103035505)